clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2794,6 +2794,19 @@ def BaseClassAddrOp : CIR_Op<"base_class_addr"> {
 // FuncOp
 //===----------------------------------------------------------------------===//
 
+// The enumeration values are not necessarily in sync with `clang::CallingConv`
+// or `llvm::CallingConv`.
+def CC_C : I32EnumAttrCase<"C", 1, "cc_c">;
+def CC_SpirKernel : I32EnumAttrCase<"SpirKernel", 75, "cc_spir_kernel">;
+def CC_SpirFunction : I32EnumAttrCase<"SpirFunction", 76, "cc_spir_function">;
```

Comment:
Not sure about the names in asm. The func op has many keywords right after the mnemonic. We do not have hint to tell "this is a calling convention keyword". Meanwhile it's hard to name the short "c" case. Therefore the prefix `cc_` was prepended accordingly.

Reply:
> We do not have hint to tell "this is a calling convention keyword"

Why not add this hint and design the assembly like:

```mlir
cir.func @foo() cc(c)
cir.func @foo() cc(spir_kernel)
cir.func @foo() cc(spir_function)
```

Reply:
Thanks for pointing it out! That's also a very nice option. Other FYI:

* We already have assembly format like `cir.func internal private @_ZL4funcv() extra(#fn_attr)`. Here `internal` is linkage and `private` is symbol visibility. (it's generated by a static function)
* I'm assuming the postpositional attribute here is in order to align with our existing form `extra(#fn_attr)`. But we may also consider that the conventional position for call conv is always before the name, no matter in the source code, LLVM MLIR dialect or LLVM IR.

Two more schemes here:
```mlir
cir.func callconv(c) @foo()
cir.func callconv(spir_kernel) @foo()
cir.func callconv(spir_function) @foo()
```

```mlir
cir.func callconv_c @foo()
cir.func spir_kernel @foo()
cir.func spir_function @foo()
```

Reply:
> Two more schemes here:

I'm down for the first one. We have attributes that specifies what "kind" the function is:

```mlir
cir.func coroutine @foo()
```

The `spir_kernel` and `spir_function` looks too similar to these kind of attributes and it's better to be explicit about their actual usage.

Reply:
- +1 for having the CC before the symbol name.
- LLVM uses `ccc` for the C calling convention. They use the no-separator-'cc'-suffix for others as well, though not for `spir_function` and `spir_kernel`.

> The spir_kernel and spir_function looks too similar to these kind of attributes and it's better to be explicit about their actual usage.

- I get where you're coming from, but I'd argue that the assembly format is not user-facing/only consumed by computers and compiler devs, so as long as it can be parsed unambiguously, wouldn't it be fine?
- I think wrapping the keyword in `cc(...)`, `cconv(...)`, `callconv(...)`, etc. is also fine, but then we probably should wrap linkage and visibility as well.

Reply:
I'm more driven towards @Lancern  idea:
```
cir.func @foo() cc(c)
cir.func @foo() cc(spir_kernel)
cir.func @foo() cc(spir_function)
```

My thoughts:
- As someone always looking into CIR output, I rather not have to wait more into finding the symbol. 
- `visibility(internal)` is quite a big name, so I think it justifies being different. Adding a `linkage(...)` would also be fair, but I feel like linkage information is something you usually also want in the face. I'm also not that worried about being super consistent if there are good reasons to deviate. `cc` is so tiny that begs to be used hehe.
- I also think that the most common CC should not be printed (i.e. we should have a default) in order not to look at that when we don't need to.

Reply:
Thank you for all the suggestions. Updated accordingly.